core/arch/riscv64/riscv64.asm

```diff
@@ -61,31 +61,29 @@ GLOBAL_LABEL(cpuid_supported:)
 GLOBAL_LABEL(call_switch_stack:)
         /* Init the stack. */
         addi     sp, sp, -32
-        /* Use two callee-save regs to call func. */
-        sd       ra, 24 (sp)
-        sd       s0, 16 (sp)
-        sd       s1, 8 (sp)
-        sd       s2, 0 (sp)
+        /* Use two callee-saved regs to call func. */
+        sd       ra, 16(sp)
```

Comment:
nit: since ra is not callee-saved, it's a little bit clearer if `sd ra, 16(sp)` is moved before the comment above (which references callee-saved regs).

Reply:
Yes, moved.